xbmc/cores/VideoPlayer/DVDCodecs/Video/VAAPI.cpp

```diff
@@ -188,6 +198,11 @@ bool CVAAPIContext::CreateContext()
     return false;
   }
 
+#if VA_CHECK_VERSION(1, 0, 0)
+  vaSetErrorCallback(m_display, VaErrorCallback, nullptr);
```

Comment:
I see two potential issues:

1) Potential crash on exit if log has gone away and vaapi fires an error/info

2) What are the downsides of logging info? This generates additional load and can block the calling thread (lock involved). I see no advantage of logging info in normal operation.

Reply:
1. Potential fix: Have a global handler for this that counts how many `CVAAPIContext` are instantiated and resets the callbacks to `nullptr` when the last one is destroyed.
2. Valid point, but a) vaapi info logging is very brief (usually limited to driver loading at init) and b) very useful because you see which driver it is trying to load, which otherwise ends up on the terminal

Reply:
ah it's `static`, even better

Reply:
Why would logging go away?

Reply:
why don't you reset the callbacks in CVAAPIContext::DestroyContext